summary_workflows/quantification/quantification_dockers/q_metrics/compute_metrics.py

```diff
@@ -52,53 +54,55 @@ def compute_metrics(participant_input, gold_standards_dir, challenge_ids, partic
-            ####################
-            # Only calculate metric for first window size.
-            if window == windows[0]:
-                match_with_gt_and_pd = matchPAS.match_with_gt(participant_input,gold_standard,window, "with_unmatched_GT_and_PD")
-                merged_bed_df_wpd, _ = match_with_gt_and_pd[0], match_with_gt_and_pd[1]
-                normalised_df = matchPAS.relative_pas_usage(merged_bed_df_wpd, genome)
-                correlation_Pearson_rel_use = matchPAS.corr_Pearson_with_gt(normalised_df)
+                # METRIC: Matched sites
+                ########################
+                # metric on the number of matched sites
+                match_with_gt_run = matchPAS.match_with_gt(f_PD=participant_input, f_GT=gold_standard, 
+                    window=window, return_df_type=return_df_type)
+                merged_bed_df, expression_unmatched = match_with_gt_run[0], match_with_gt_run[1]
+
+                # Key: exact name of metric as it appears in specification
+                metric_name = f"Expression_non-matched-PAS_{window}nt"
+                # Value: List of [variable_holding_metric, std_err]
+                metrics[metric_name] = [expression_unmatched, 0]
+
+
+                # METRIC: correlation coefficient
+                #################################
+                # metric on correlation coefficient
+                correlation_pearson = matchPAS.corr_Pearson_with_gt(merged_bed_df)
                 # Key: exact name of metric as it appears in specification
-                metric_name = f"Correlation_coefficient_Pearson_relative_{window}nt"
+                metric_name = f"Correlation_coefficient_Pearson_{return_df_type}_{window}nt"
                 # Value: List of [variable_holding_metric, std_err]
-                metrics[metric_name] = [correlation_Pearson_rel_use, 0]
+                metrics[metric_name] = [correlation_pearson, 0]
                 # Spearman correlation
-                correlation_Spearman_rel_use = matchPAS.corr_Spearman_with_gt(normalised_df)
+                correlation_spearman = matchPAS.corr_Spearman_with_gt(merged_bed_df)
                 # Key: exact name of metric as it appears in specification
-                metric_name = f"Correlation_coefficient_Spearman_relative_{window}nt"
+                metric_name = f"Correlation_coefficient_Spearman_{return_df_type}_{window}nt"
                 # Value: List of [variable_holding_metric, std_err]
-                metrics[metric_name] = [correlation_Spearman_rel_use, 0]
+                metrics[metric_name] = [correlation_spearman, 0]
+
+                # METRIC: correlation coefficient of relative pas usage
+                ####################
+                # Only calculate metric for first window size.
+                if window == windows[0]:
+                    match_with_gt_and_pd = matchPAS.match_with_gt(f_PD=participant_input, f_GT=gold_standard,
```

Comment:
the variable names `match_with_gt_and_pd`, as well as `merged_bed_df_wpd` are misleading after the refactoring you did, because now this df is calculated for all three return types. Actually, couldn't you therefore even use the `merged_bed_df` from above? As far as I understand the code there is no need to re-calculate it.

Reply:
good catch! Will change